### ISSUE TITLE
Confidence of all keypoints in pose_object

### ISSUE THREAD

**Issue by nikhilchh** [wmcnally/kapao#24]
You have mentioned in the paper:
1-Human pose predictions typically contain a sparse set of keypoint confidences because pose_object doesn't provide keypoint confidences by design. 
2- If we reply solely on keypoint objects then we miss a lot of keypoints.

So what is the challenge in adding another 17 values in the object representation. So instead of 34 (x,y) it would become 51 (x,y,c) values.

**Comment by wmcnally** [comment]
> So what is the challenge in adding another 17 values in the object representation. So instead of 34 (x,y) it would become 51 (x,y,c) values.

You could - it would require adding another loss component and balancing it. Using a low keypoint object confidence threshold (as suggested in the paper) actually works quite well, though. See the red light green light demo. I plot incomplete poses there using keypoint objects only.

**Comment by nikhilchh** [comment]
That's a good point. Should be good enough. And clearly its a trade-off. 

So this is what happens by design when we only use keypoint object :
1- We use pose object only for gouping keypoints. Right ?
2- We would miss hidden keypoints if we don't use kp from pose objects. Because: Pose object in my understanding is learning the concept of human skeleton and it will be probably be good at estimating hidden keypoints because it makes network understand the concept of human skeleton structure(graph). Keypoint objects on the other hand would be good at localizing visible keypoints. Keypoints object is more about focusing on local patch of the image. Correct me if I am wrong in understanding this. Or Do you think keypoint object can also localize hidden keypoints based on context information ?

Btw are hidden keypoint treated differently in any way during the training ?

**Comment by wmcnally** [comment]
Yes you are 100% correct. However, I have found that keypoint objects sometimes pickup hidden (occluded) keypoints as well. If you are familiar with the coco format, I treat all keypoints with visibility flag > 0 the same.

**Comment by nikhilchh** [comment]
Ok got it. 
Thanks :)
